pytorch_lightning/utilities/parsing.py

```diff
@@ -225,12 +226,23 @@ def save_hyperparameters(
         for local_args in collect_init_args(frame, []):
             init_args.update(local_args)
 
-    if ignore is not None:
-        if isinstance(ignore, str):
-            ignore = [ignore]
-        if isinstance(ignore, (list, tuple)):
-            ignore = [arg for arg in ignore if isinstance(arg, str)]
-        init_args = {k: v for k, v in init_args.items() if k not in ignore}
+    if ignore is None:
+        ignore = set()
+    elif isinstance(ignore, str):
+        ignore = {ignore}
+    elif isinstance(ignore, (list, tuple)):
+        ignore = {arg for arg in ignore if isinstance(arg, str)}
+
+    for k in list(init_args):
+        if k in ignore:
+            del init_args[k]
+            continue
+
+        if isinstance(init_args[k], nn.Module):
```

Comment:
shall we also add an argument that the user would need to explicitly allow saving `nn.Module` otherwise drop it?

Reply:
this will save the `nn.Modules`, but will raise a warning in that case, because of checkpointing issue. I don't think we need an argument since that will only disable the warnings I guess.